packages/casadi/patches/0002-disable-size-max-check.patch

```diff
@@ -0,0 +1,54 @@
+From 2d025093decd62a69d6d400340617e9baf116858 Mon Sep 17 00:00:00 2001
+From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
+Date: Thu, 18 Jul 2024 02:08:07 +0530
+Subject: [PATCH 2/3] Disable SIZE_MAX check for Emscripten
+
+In WebAssembly (currently 32-bit), SIZE_MAX is equal to UINT_MAX, which meant
+these functions weren't being compiled, so there was no way to serialize or
+deserialize unsigned int values, leading to compile errors when these functions
+were called elsewhere. On many 32-bit systems, size_t and unsigned int are the
+same size, so SIZE_MAX == UINT_MAX is a common case.
+
+---
+ casadi/core/serializing_stream.cpp | 2 +-
+ casadi/core/serializing_stream.hpp | 4 ++--
+ 2 files changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/casadi/core/serializing_stream.cpp b/casadi/core/serializing_stream.cpp
+index f3101f84..dfc6b2e4 100644
+--- a/casadi/core/serializing_stream.cpp
++++ b/casadi/core/serializing_stream.cpp
+@@ -155,7 +155,7 @@ namespace casadi {
+       for (int j=0;j<4;++j) pack(c[j]);
+     }
+ 
+-#if SIZE_MAX != UINT_MAX
++#if SIZE_MAX != UINT_MAX || defined(__EMSCRIPTEN__)
```

Comment:
This makes me think that the check is not correct. It's testing for 64 bit systems but we also want to compile this on Emscripten which is 32 bits? I wonder if using `int32_t` and `int64_t` for the two definitions would make it possible to get rid of this entirely.

Anyways, please open an upstream PR and link it in the patch file.

Reply:
In our case, removing the check entirely is working. For an upstream PR, I can use `defined(__EMSCRIPTEN__)`.

Reply:
Removed the check entirely to allow compilation in fcce1803a678a9667db683335f7ccb21cc4cb7d4, and added a link to the upstream PR in 4a01fde7ee509ba225638485414b0e572a380616.

Reply:
Well we want to match our patches in line with what we would expect to upstream if possible, so it's probably good to keep what you had, unless you think upstream will be willing to remove the check.

Reply:
Oh I see they merged your PR, so we can backport those commits with `git format-patch` and then add a reference to the PR to the patch messages.

Reply:
On it! Do you mean I should remove the two patches I added here and replace them with a single patch generated from the merged PR?

Reply:
(The reason I kept the patches different here vs. upstream was because I didn't wish to revert the related commit through my patch)